### PR TITLE
fix: Exit with code 1 when nyc doesn't know what to do.

### PR DESCRIPTION
**Pull request by coreyfarrell**
When nyc is not told to do anything this is an error.  We already
printed the help message to stderr, now we exit with code 1 to indicate
that nyc exited without doing anything.

Add a test to cover this edge case and verify that both `nyc` and `nyc
--help` produce output as expected to stdout and stderr, exit with the
proper code.

### CODE REVIEW

File: test/nyc-integration.js [hunk 1]
```diff
@@ -1079,6 +1079,40 @@ describe('the nyc cli', function () {
...
+      }
+      proc.stdout.on('data', chunk => {
+        msgs[type].out += chunk
+      })
+      proc.stderr.on('data', chunk => {
+        msgs[type].err += chunk
+      })
+      proc.on('close', code => {
+        code.should.equal(expectCode)
+        waitingFor--
```
Comment (JaKXz) [inline test/nyc-integration.js]: While I think this is clever, I feel like it could be done simpler. I've been wanting to introduce https://npm.im/await-spawn for a while, what do you think of it here?

Comment (coreyfarrell) [reply test/nyc-integration.js]: I think that would be cool once we can use async/await (nyc@15).  As long as we're still supporting node.js 6 and not transpiling test/*.js I'd rather leave it as is.

Comment (JaKXz) [reply test/nyc-integration.js]: It returns a promise so we don't have to use the new syntax just yet. Just:

```js
return spawn(...)
  .then(() => {
    //asserts
  })
```

Comment (coreyfarrell) [reply test/nyc-integration.js]: I'm OK with using await-spawn for these new tests but I don't think we should do a mass change to `return spawn(...).then(...)`.  I'd much rather we just leave existing tests alone until we can use async/await sytax.

Comment (coreyfarrell) [reply test/nyc-integration.js]: Actually it looks like `await-spawn` won't let us see both stdout and stderr.  I think our testing would be better served by something which mimics the API of `spawnSync`, just with promises.  So `spawn(...).then(({stdout, stderr, status}) => { ... })`.  This is something which is currently lacking IMO, we should be checking for empty stderr in most tests (including when the exit code is 0).

Comment (coreyfarrell) [reply test/nyc-integration.js]: Any feelings about using [execa](https://github.com/sindresorhus/execa#readme) in tests?  It's actually already installed via yargs so although it's a bigger module it wouldn't even increase our node_modules size.

Comment (JaKXz) [reply test/nyc-integration.js]: Sure! execa looks like a better fit, I agree :)

Comment (coreyfarrell) [reply test/nyc-integration.js]: Very nice.

File: test/nyc-integration.js [hunk 2]
```diff
@@ -14,6 +14,8 @@ const rimraf = require('rimraf')
 const makeDir = require('make-dir')
 const spawn = require('child_process').spawn
 const si = require('strip-indent')
+// BUGBUG: when upgrading execa to 2.x stripEof is renamed to stripFinalNewline.
```
Comment (JaKXz) [inline test/nyc-integration.js]: should we not upgrade because of this? what's wrong?

Comment (coreyfarrell) [reply test/nyc-integration.js]: 2.x isn't released yet and will require node.js 8.

File: test/nyc-integration.js [hunk 3]
```diff
@@ -1079,6 +1081,25 @@ describe('the nyc cli', function () {
...
   })
 
+  it('help shows to stderr when main command doesn\'t know what to do', () => {
+    const opts = {
+      cwd: fixturesCLI,
+      env,
+      stripEof: false
+    }
+    return Promise.all([
+      execa(process.execPath, [bin, '--help'], opts),
```
Comment (bcoe) [inline test/nyc-integration.js]: any reason we can't just use `spawnSync` here, I've been trying to use less dependencies when possible.

Comment (coreyfarrell) [reply test/nyc-integration.js]: execa is already a dependency (it's installed by yargs).  I'd like to leave the door open for parallel testing which can never happen if we use sync methods everywhere.

File: test/nyc-integration.js (hunk 2 above)
Comment (coreyfarrell) [reply test/nyc-integration.js]: @bcoe has requested that we not use execa, for now I'm just going to use spawnSync.

Comment (bcoe) [reply test/nyc-integration.js]: :+1: love to use Node's built-ins where possible, and it felt a bit weird that we were adding a caveat:

```
// BUGBUG: 
```

unrelated to our test; thanks for entertaining me being curmudgeonly.

File: test/nyc-integration.js (hunk 3 above)
Comment (JaKXz) [reply test/nyc-integration.js]: ^ agree with @coreyfarrell here
